configure.ac

```diff
@@ -1554,80 +1750,76 @@ dnl ***************************************************************************
 dnl libesmtp headers/libraries
 dnl ***************************************************************************
 if test "x$enable_smtp" != "xno" && test "x$with_libesmtp" != "xno"; then
-	libesmtp="yes"
-	if test "x$with_libesmtp" != "xyes" && test "x$with_libesmtp" != "x"; then
-		CPPFLAGS_SAVE="$CPPFLAGS"
-		LDFLAGS_SAVE="$LDFLAGS"
-		CPPFLAGS="$CPPFLAGS -I$with_libesmtp/include"
-		LDFLAGS="$LDFLAGS -L$with_libesmtp/lib"
-		AC_CHECK_HEADER(libesmtp.h, [LIBESMTP_CFLAGS="-I$with_libesmtp/include"
-					     LIBESMTP_LIBS="-L$with_libesmtp/lib -lesmtp"], [libesmtp=no])
-		CPPFLAGS="$CPPFLAGS_SAVE"
-		LDFLAGS="$LDFLAGS_SAVE"
-	else
-		PKG_CHECK_MODULES(LIBESMTP, libesmtp-1.0, libesmtp="yes", libesmtp="no")
-		if test "x$libesmtp" = "xno"; then
-			AC_MSG_CHECKING(for libESMTP with libesmtp-config)
-			if libesmtp-config --version >/dev/null 2>&1; then
-				AC_MSG_RESULT(yes)
-				LIBESMTP_CFLAGS="`libesmtp-config --cflags`"
-				LIBESMTP_LIBS="`libesmtp-config --libs`"
-				libesmtp="yes"
-			else
-				AC_MSG_RESULT(no)
-				libesmtp=no
-			fi
+    libesmtp="yes"
+    if test "x$with_libesmtp" != "xyes" && test "x$with_libesmtp" != "x"; then
+        CPPFLAGS="$CPPFLAGS -I$with_libesmtp/include"
+        LDFLAGS="$LDFLAGS -L$with_libesmtp/lib"
+        AC_CHECK_HEADERS(libesmtp.h, [LIBESMTP_CFLAGS="-I$with_libesmtp/include"
+                        LIBESMTP_LIBS="-L$with_libesmtp/lib -lesmtp"], [libesmtp=no])
+        CPPFLAGS="$CPPFLAGS_SAVE"
+        LDFLAGS="$LDFLAGS_SAVE"
```

Comment:
Would'n `SAFE_C_CHECK_END` restore these? (multiple occurrences on this one)

Reply:
same as above, safer to restore inside if/else/switch/etc. to eliminate future issues